File: OffByOne/src/OffByOne.LanguageCore/Ast/Literals/BooleanLiteral.cs
namespace OffByOne.LanguageCore.Ast.Literals
{
    using OffByOne.LanguageCore.Ast.Literals.Base;
    using OffByOne.LanguageCore.Visitors.Contracts;

    public class BooleanLiteral : Literal
    {
        public BooleanLiteral(bool value)
        {
            this.Value = value;
        }

        public BooleanLiteral(string value)
            : this(bool.Parse(value))
        {
        }

        public bool Value { get; private set; }

        public override TResult Accept<TResult, TContext>(
            ILiteralVisitor<TResult, TContext> visitor,
            TContext context)
        {
            return visitor.Visit(this, context);
        }
    }
}

File: vanVlaanderenDerksen/QLJava/src/sc/ql/model/expressions/literals/BooleanLiteral.java
package sc.ql.model.expressions.literals;

import sc.ql.model.Expression;
import sc.ql.model.visitors.ExpressionVisitor;

public class BooleanLiteral extends Expression {
	private final Boolean value;
	
	public BooleanLiteral(Boolean value) {
		this.value = value;
	}
	
	public Boolean getValue() {
		return this.value;
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
}
